nio-impl/src/main/java/org/xnio/nio/QueuedNioTcpServer.java

```diff
@@ -401,14 +418,16 @@ public SocketAddress getLocalAddress() {
 
     public void suspendAccepts() {
         handle.suspend(0);
+        suspendedDueToWatermark = false;
     }
 
     public void resumeAccepts() {
         handle.resume(SelectionKey.OP_ACCEPT);
+        suspendedDueToWatermark = false;
     }
 
     public boolean isAcceptResumed() {
-        return handle.isResumed(SelectionKey.OP_ACCEPT);
+        return handle.isResumed(SelectionKey.OP_ACCEPT) || suspendedDueToWatermark;
```

Comment:
This method should only reflect the status of user suspention, not connection limit suspension.